ablySpec/TestUtilities.swift

```diff
@@ -414,6 +414,13 @@ class TestProxyTransport: ARTWebSocketTransport {
         super.receive(msg)
     }
 
+    func closeAbruptly() {
+        // Inject a ProtocolMessage.CLOSE
```

Comment:
Agreed, this is completely the wrong behaviour.  We want the transport `on(closed)` event equivalent to be triggered by the test suite so that the client library thinks the transport was abruptly terminated.